### PR TITLE
Added `fetch:plain:file` action to fetch a single file

### LINKED ISSUE #15388: 🚀 Feature: Make `fetch:plain` to allow fetching a single file instead of directory
### 🔖 Feature description

Currently `fetch:plain` only allows to fetch directories, but for some cases it's useful to fetch individual files.

### 🎤 Context

We are utilizing scaffolds to automate service creation and as part of that we also open few MRs to a separate repository. We have a `CODEOWNERS` file in that repository and ideally we want to reduce manual actions and update that file with the team that has created a service.
I'm looking for a workflow like this:
```
    - id: fetch_codeowners
      name: Fetch CODEOWNERS from IaC
      action: fetch:plain
      input:
        url: https://gitlab.com/.../-/blob/main/CODEOWNERS
        targetPath: external/iac

    - id: update_codeowners
      name: Update CODEOWNERS
      action: miro:update-codeowners
      input:
        path: external/iac/CODEOWNERS
```

### ✌️ Possible Implementation

_No response_

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, but I'm happy to collaborate on a PR with someone else

### PR DESCRIPTION
**Pull request by gavlyukovskiy**
## Hey, I just made a Pull Request!

This PR adds `fetch:plain:file` action as discussed in the original issue #15388. Main reason for this separate action is that `fetch:plain` only allows to fetch directories:
```yaml
    - id: fetch-docs
      name: Fetch Docs
      action: fetch:plain
      input:
        targetPath: ./community
        url: https://github.com/backstage/community/tree/main/backstage-community-sessions
```
but not a single file (i.e. https://github.com/backstage/community/tree/main/backstage-community-sessions/assets/Backstage%20Community%20Sessions.png).

Alternatively one could fetch directory and remove all other files, but for some cases that's not a trivial thing to do - our main use-case is to fetch `CODEOWNERS` file from root of a large repository.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

Fixes #15388

### CODE REVIEW

File: plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/helpers.ts [hunk 1]
```diff
@@ -35,43 +35,85 @@ export async function fetchContents(options: {
...
     await fs.ensureDir(outputPath);
     await res.dir({ targetDir: outputPath });
   }
 }
+
+/**
+ * A helper function that reads the content of a single file from the given URL.
+ * Can be used in your own actions, and also used behind fetch:plain:file
+ *
+ * @public
```
Comment (jhaals) [inline plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/helpers.ts:58]:
```suggestion
 * Can be used in your own actions, and also used behind `fetch:plain:file`
```
We are not exporting this function so no need to mark it public :)

File: .changeset/fuzzy-wombats-search.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-scaffolder-backend': minor
+---
+
+Added `fetch:plain:file` action to fetch a single file
```
Comment (jhaals) [inline .changeset/fuzzy-wombats-search.md:5]:
```suggestion
Added `fetch:plain:file` action to fetch a single file, this action is also added to the list of built-in actions.
```

**Review by jhaals — DISMISSED** [review]
Nice work with this! ⭐ It looks good to me, tiny tweaks.

Requesting one quick 👁️ from @benjdlambert as well

File: plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/helpers.test.ts [hunk 3]
```diff
@@ -50,7 +51,7 @@ describe('fetchContent helper', () => {
     outputPath: os.tmpdir(),
   };
 
-  it('should reject absolute file locations', async () => {
+  it('fetch contents should reject absolute file locations', async () => {
```
Comment (benjdlambert) [inline plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/helpers.test.ts:54]: I'd wrap all these up in a separate `describe` block instead. as `it fetch contents` doesn't really read very well.

```ts
describe('fetch contents', () => {
  it('rejects absolute file locations')
})
```

File: plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/helpers.test.ts [hunk 4]
```diff
@@ -116,7 +117,92 @@ describe('fetchContent helper', () => {
       outputPath: 'foo',
       fetchUrl: 'https://github.com/backstage/foo',
     });
-    expect(fs.ensureDir).toHaveBeenCalled();
+    expect(fs.ensureDir).toHaveBeenCalledWith('foo');
     expect(dirFunction).toHaveBeenCalledWith({ targetDir: 'foo' });
   });
+
+  it('fetch file should reject absolute file locations', async () => {
```
Comment (benjdlambert) [inline plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/helpers.test.ts:124]: `describe('fetch file')`

**Review by benjdlambert — CHANGES_REQUESTED** [review]
Looking pretty good - just some small bits.

File: plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/helpers.test.ts (hunk 3 above)
Comment (gavlyukovskiy) [reply plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/helpers.test.ts]: It looks much nicer, thanks for the suggestion.

**Comment by gavlyukovskiy** [comment]
@benjdlambert @jhaals I have applied your suggestions

**Comment by regicsolutions** [comment]
@gavlyukovskiy I gave this action a try but the downloaded file doesn't seem to be placed into the workspace/temp path, was hoping to download a file before running the `fetch:template` action which targets the  `./skeleton `

**Pull request merged**